budget/app.py

```diff
@@ -1,14 +1,176 @@
 import json
 import logging
 
+import boto3
+from botocore.exceptions import ClientError
+import requests
+
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
+# TODO configure this URL?
+SYNAPSE_TEAM_MEMBER_URL = 'https://repo-prod.prod.sagebase.org/repo/v1/teamMembers'
+
+# TODO pull this config from S3?
+USER_BUDGET_BY_TEAM = {
+  '12345': {
+    'amount': 2000,
+    'period': 'annual'
+  },
+  '67890': {
+    'amount': 100,
+    'period': 'annual'
+  }
+}
+
+
+def get_client(service):
+  return boto3.client(service)
+
+
+def get_users_by_team(teams):
+  '''Get users from synapse teams
+
+  Returns a dictionary of users with a list of their team memberships
+  '''
+  teams_by_user_id = {}
+  for team_id in teams:
+    synapse_url = f'{SYNAPSE_TEAM_MEMBER_URL}/{team_id}'
+    response = requests.get(synapse_url)
+    if response.ok:
+      results = json.loads(response.text)['results']
+      user_ids = [result['member']['ownerId'] for result in results if not result['isAdmin']]
```

Comment:
why omit the 'admin' (aka 'team manager')?

Reply:
because making a budget doesn't apply to them

Reply:
Are you sure?  Couldn't someone both be a Service Catalog user and a team manager?

Reply:
TLDR: yes, I'll add that to the CM SOP

Budgets are for external users. Team managers are internal users. This budget making app specifically will only look at external teams and there is an expectation that the team members (not admins) occur in only one of these teams. Internal user-admins will probably occur in multiple external teams as managers.